### PR TITLE
[HER-117] Add filters to detections page

### PR DESCRIPTION
**Pull request by ConstantinescuAndrei**
## Descriptions
- add the filter component on the detections page
- call the cameras query

## Related
[HER-117](https://nerds-interns.atlassian.net/jira/software/projects/HER/boards/1?selectedIssue=HER-117)

### CODE REVIEW

File: src/detections/filters/accordion-actions/add-filter.jsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { Button } from '@material-ui/core'
+import AddIcon from '@material-ui/icons/Add'
+
+import { useIsDisabled } from 'detections/filters/hooks/use-is-disabled'
+import { useAddFilterComponent } from 'detections/filters/hooks/use-add-filter-component'
+
+export const AddFilter = () => <Button fullWidth variant={'contained'} color={'primary'}
```
Comment (miutamihai) [inline src/detections/filters/accordion-actions/add-filter.jsx:8]: Whenever we've got to pass so many props that we can't fit them on a single line, we should extract them into a separate `useProps` hook.

File: src/detections/filters/accordion-actions/filter-detections.jsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+import { Button } from '@material-ui/core'
+import SearchIcon from '@material-ui/icons/Search'
+
+export const FilterDetections = () => <Button fullWidth
```
Comment (miutamihai) [inline src/detections/filters/accordion-actions/filter-detections.jsx:5]: Same here.

File: src/detections/filters/accordion-details/cameras/select.jsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import { Select as MuiSelect } from '@material-ui/core'
+import { useSelector } from 'react-redux'
+
+import { MenuItem } from 'detections/filters/accordion-details/cameras/menu-item'
+
+export const Select = () => {
+  const { items } = useSelector(({ cameras }) => cameras.get.data)
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/cameras/select.jsx:8]: Let's make sure to fix this :)))
image

File: src/detections/filters/accordion-details/class/index.jsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import { FilterWrapper } from 'detections/filters/accordion-details/filter-wrapper'
+import { RemoveButton } from 'detections/filters/accordion-details/remove-button'
+import { Input } from 'detections/filters/accordion-details/class/input'
+import { FilterTitle } from 'detections/filters/accordion-details/filter-title'
+
+export const Class = index => <FilterWrapper>
+  <FilterTitle title={'Class'} />
+  <Input />
+  <Grid container item xs={1} sm={2} md={2} lg={2} xl={2} justify={'flex-end'} alignContent={'center'}>
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/class/index.jsx:12]: Grids cannot be both an item and a container at the same time. Let's split this into separate grids.

The same applies for the others as well.

File: src/detections/filters/context.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import React, { useContext as useReactContext, useState } from 'react'
+
+export const Context = React.createContext({})
+
+export const useContext = () => useReactContext(Context)
+
+export const useDefaultContext = () => {
+  const [filters, setFilters] = useState([])
+  const [selectItems, setSelectItems] = useState(['Class', 'Score', 'Timestamp', 'Cameras'])
```
Comment (miutamihai) [inline src/detections/filters/context.js:9]: Let's rename this to something like `filterVariants`

File: src/detections/filters/accordion-details/filter/wrapper.jsx [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import { Grid } from '@material-ui/core'
+import PropTypes from 'prop-types'
+
+import { useStyles } from 'detections/filters/accordion-details/filter/style'
+
+export const Wrapper = ({ children }) => <Grid container justify={'space-between'} alignItems={'center'}
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/filter/wrapper.jsx:7]: Same here.

File: src/detections/filters/accordion-details/remove-button/index.jsx [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import React from 'react'
+import HighlightOffIcon from '@material-ui/icons/HighlightOff'
+import PropTypes from 'prop-types'
+
+import { useRemoveFilter } from 'detections/filters/hooks/use-remove-filter'
+import { useStyles } from 'detections/filters/accordion-details/remove-button/style'
+
+export const RemoveButton = ({ index, type }) => <HighlightOffIcon onClick={useRemoveFilter(index, type)}
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/remove-button/index.jsx:8]: Same here.

File: src/detections/filters/accordion-details/score/index.jsx [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+
+import { Input } from 'detections/filters/accordion-details/score/input'
+import { RemoveButton } from 'detections/filters/accordion-details/remove-button'
+import { FilterWrapper } from 'detections/filters/accordion-details/filter-wrapper'
+import { FilterTitle } from 'detections/filters/accordion-details/filter-title'
+
+export const Score = index => <FilterWrapper>
+  <FilterTitle title={'Score'} />
+  <Input />
+  <Grid container item xs={1} sm={2} md={2} justify={'flex-end'} alignContent={'center'}>
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/score/index.jsx:12]: Same here.

File: src/detections/filters/hooks/use-remove-filter.js [hunk 9]
```diff
@@ -0,0 +1,13 @@
+import { useCallback } from 'react'
```
Comment (miutamihai) [inline src/detections/filters/hooks/use-remove-filter.js:1]: Let's leave an empty line here.

File: src/detections/filters/hooks/use-remove-filter.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+import { useContext } from 'detections/filters/context'
+
+const deleteFilter = index => filters => filters.filter((item, filterIndex) => filterIndex !== index)
+
+export const useRemoveFilter = (index, type) => {
+  const { setFilters, selectItems, setSelectItems } = useContext()
+
+  return useCallback(() => {
+    type !== 'filter' && setSelectItems([...selectItems, type])
```
Comment (miutamihai) [inline src/detections/filters/hooks/use-remove-filter.js:10]: Shouldn't this be an empty string?

File: src/detections/index.jsx [hunk 11]
```diff
@@ -1,11 +1,18 @@
...
+import { Wrapper } from 'detections/wrapper'
+import { Filters } from 'detections/filters'
+import { useFetchCameras } from 'detections/hooks/use-fetch-cameras'
 
-export const Detections = () => <Grid container>
-  <Content />
-  <Table />
-</Grid>
+export const Detections = () => {
+  useFetchCameras()
```
Comment (miutamihai) [inline src/detections/index.jsx:10]: Let's use a `<Behaviour/>` component here.

**Review by miutamihai — CHANGES_REQUESTED** [review]

File: src/detections/filters/accordion-actions/add-filter.jsx (hunk 1 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-actions/add-filter.jsx]: done

File: src/detections/filters/accordion-actions/filter-detections.jsx (hunk 2 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-actions/filter-detections.jsx]: done

File: src/detections/filters/accordion-details/cameras/select.jsx (hunk 3 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/cameras/select.jsx:8]: done

File: src/detections/filters/accordion-details/class/index.jsx (hunk 4 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/class/index.jsx]: done

File: src/detections/filters/context.js (hunk 5 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/context.js]: done

File: src/detections/filters/accordion-details/remove-button/index.jsx (hunk 7 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/remove-button/index.jsx]: done

File: src/detections/filters/accordion-details/filter/wrapper.jsx (hunk 6 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/filter/wrapper.jsx]: done

File: src/detections/filters/accordion-details/score/index.jsx (hunk 8 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/score/index.jsx]: done

File: src/detections/filters/hooks/use-remove-filter.js (hunk 9 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/hooks/use-remove-filter.js:1]: done

File: src/detections/filters/hooks/use-remove-filter.js (hunk 10 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/hooks/use-remove-filter.js]: done

File: src/detections/index.jsx (hunk 11 above)
Comment (ConstantinescuAndrei) [reply src/detections/index.jsx]: done

File: src/detections/filters/accordion-details/cameras/menu-item.jsx [hunk 12]
```diff
@@ -0,0 +1,15 @@
+import React, { useState } from 'react'
+import { MenuItem as MuiMenuItem, Checkbox } from '@material-ui/core'
+import PropTypes from 'prop-types'
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/cameras/menu-item.jsx:3]: Let's leave an empty line here.

File: package.json [hunk 13]
```diff
@@ -4,10 +4,13 @@
   "private": true,
   "dependencies": {
     "@apollo/client": "3.4.3",
+    "@date-io/date-fns": "1.x",
```
Comment (miutamihai) [inline package.json:7]: Let's use a fixed version here.

File: src/detections/filters/accordion-details/cameras/index.jsx [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
+import { FilterWrapper } from 'detections/filters/accordion-details/filter-wrapper'
+import { RemoveButton } from 'detections/filters/accordion-details/remove-button'
+import { FilterTitle } from 'detections/filters/accordion-details/filter-title'
+import { Content } from 'detections/filters/accordion-details/cameras/content'
+
+export const Cameras = index => <FilterWrapper>
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/cameras/index.jsx:8]: Let's make sure we pass props the way props are meant to be passed: as fields inside of an object.

This applies to all files touched in this pr.

File: src/detections/filters/accordion-details/filter/wrapper.jsx [hunk 15]
```diff
@@ -0,0 +1,11 @@
...
+import { Grid } from '@material-ui/core'
+import PropTypes from 'prop-types'
+
+import { useProps } from 'detections/filters/accordion-details/filter/use-props'
+
+export const Wrapper = ({ children }) => <Grid {...useProps()}>
+  {children}
+</Grid>
+
+Wrapper.propTypes = { children: PropTypes.element.isRequired }
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/filter/wrapper.jsx:11]: You sure this wrapper receives only one element as its children?

File: src/detections/filters/accordion-details/hooks/use-pick-cameras-template.js [hunk 16]
```diff
@@ -0,0 +1,9 @@
+import { useDisplayCase } from 'detections/filters/accordion-details/hooks/use-display-case'
+import { useCamerasTemplate } from 'detections/filters/accordion-details/hooks/use-cameras-template'
+
+export const usePickCamerasTemplate = () => {
+  const displayCase = useDisplayCase().true || 'loading'
+  console.log(useDisplayCase().true)
```
Comment (miutamihai) [inline src/detections/filters/accordion-details/hooks/use-pick-cameras-template.js:6]: Let's remove this log

File: src/detections/filters/wrapper.jsx [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+import { Grid } from '@material-ui/core'
+import PropTypes from 'prop-types'
+
+import { useProps } from 'detections/filters/use-props'
+
+export const Wrapper = ({ children }) => <Grid {...useProps()}>
+  {children}
+</Grid>
+
+Wrapper.propTypes = { children: PropTypes.element.isRequired }
```
Comment (miutamihai) [inline src/detections/filters/wrapper.jsx:11]: Same here.

Comment (miutamihai) [reply src/detections/filters/wrapper.jsx:11]: And all components receiving the `Children` prop.

File: src/detections/wrapper.jsx [hunk 18]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'
+import { Grid } from '@material-ui/core'
+import PropTypes from 'prop-types'
+
+export const Wrapper = ({ children }) => <Grid container direction={'column'}
```
Comment (miutamihai) [inline src/detections/wrapper.jsx:5]: Multiline props alert.

**Review by miutamihai — CHANGES_REQUESTED** [review]
image
This doesn't look good... Remove the lock file and generate it again.

File: src/detections/filters/accordion-details/cameras/menu-item.jsx (hunk 12 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/cameras/menu-item.jsx:3]: done

File: package.json (hunk 13 above)
Comment (ConstantinescuAndrei) [reply package.json]: done

File: src/detections/filters/accordion-details/cameras/index.jsx (hunk 14 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/cameras/index.jsx]: done

File: src/detections/filters/accordion-details/filter/wrapper.jsx (hunk 15 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/filter/wrapper.jsx]: done

File: src/detections/filters/accordion-details/hooks/use-pick-cameras-template.js (hunk 16 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/accordion-details/hooks/use-pick-cameras-template.js]: done

File: src/detections/filters/wrapper.jsx (hunk 17 above)
Comment (ConstantinescuAndrei) [reply src/detections/filters/wrapper.jsx]: done

File: src/detections/wrapper.jsx (hunk 18 above)
Comment (ConstantinescuAndrei) [reply src/detections/wrapper.jsx]: done

**Review by miutamihai — APPROVED** [review]

**Review by RaduBuciuceanu — APPROVED** [review]
Sexy!

**Pull request merged**